pycheribuild/projects/project.py

```diff
@@ -1645,6 +1645,44 @@ def update(self, current_project: "Project", *, src_dir: Path, base_project_sour
             run_command("git", "stash", "pop", cwd=src_dir, print_verbose_only=True)
 
 
+class SubversionRepository(SourceRepository):
+    def __init__(self, url, *, default_branch: str = None):
+        self.url = url
+        self._default_branch = default_branch
+
+    def ensure_cloned(self, current_project: "Project", src_dir: Path, **kwargs):
+        if (src_dir / ".svn").is_dir():
+            return
+
+        if current_project.config.skip_clone:
+            current_project.fatal("Sources for", str(src_dir), " missing!")
+            return
+
+        assert isinstance(self.url, str), self.url
+        assert not self.url.startswith("<"), "Invalid URL " + self.url
```

Comment:
I couldn't remember why I added that assert in 210264525a61c9e75b13a0d7d26782c9c021113e. I believe the problem was that I had accidentally used `str(<instance of GitRepository>)`. And the default python str() is `<$CLASS object at 0x....>`. Not sure this assert is required.

Reply:
I had guessed that was your reason which is why I kept it when copying from GitRepository.  I don't mind dropping it though if you'd rather me do so.

Reply:
Seems fine to keep it